internal/putil/putil.go

```diff
@@ -106,6 +109,8 @@ func MakeKnown(v resource.PropertyValue) resource.PropertyValue {
 // 2. It doesn't panic when computed values are present.
```

Comment:
This comment is not true anymore, is it? `contract.Assert` panics.

Reply:
On second thought, the whole method doesn't seem to make sense anymore. When secrets and computed values are not allowed, it's just vanilla DeepEquals, isn't it?

Reply:
> This comment is not true anymore, is it? contract.Assert panics.

The comment is still true. `foldOutputValue` normalizes both secret (`.IsSecret()`) and computed (`.IsComputed()`) values into outputs (`.IsOutput()`) values to simplify the comparison. 

👇 is still true (and does not panic):

```go
putil.DeepEquals(
    resource.MakeComputed(resource.MakeSecret(resource.NewProperty(true))),
    resource.MakeSecret(resource.MakeComputed(resource.NewProperty(true))),
)
```

`foldOutputValue` never returns top level `resource.Secret` or `resource.Computed` values, so the asserts are valid.